pkg/dtclient/dynatrace_client.go

```diff
@@ -146,15 +147,18 @@ func (dc *dynatraceClient) setHostCacheFromResponse(response []byte) error {
 
 	for _, info := range hostInfoResponses {
 		hostInfo := hostInfo{entityID: info.EntityID}
-
-		v := info.AgentVersion
-		if v == nil {
-			continue
-		}
-
-		hostInfo.version = fmt.Sprintf("%d.%d.%d.%s", v.Major, v.Minor, v.Revision, v.Timestamp)
-		for _, ip := range info.IPAddresses {
-			dc.hostCache[ip] = hostInfo
+		nz := info.NetworkZoneID
+
+		if nz == networkZone || nz == "default" {
```

Comment:
If you set the network zone, we should ignore hosts that don't have a network set.

```suggestion
		if nz == networkZone || networkZone == "" {
```